File: src/main/java/com/wgzhao/addax/admin/model/EtlColumnPk.java
package com.wgzhao.addax.admin.model;

import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

@Setter
@Getter
public class EtlColumnPk
        implements Serializable
{
    private long tid;
    private String columnName;

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        EtlColumnPk that = (EtlColumnPk) o;
        return tid == that.tid && columnName.equals(that.columnName);
    }

    @Override
    public int hashCode() {
        return Long.hashCode(tid) * 31 + columnName.hashCode();
    }
}

File: src/main/kotlin/com/wgzhao/addax/admin/model/EtlColumnPk.kt
package com.wgzhao.addax.admin.model

import java.io.Serializable


class EtlColumnPk

    : Serializable {
    private val tid: Long = 0
    private val columnId = 0

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (o == null || javaClass != o.javaClass) return false
        val that = o as EtlColumnPk
        return tid == that.tid && columnId == that.columnId
    }

    override fun hashCode(): Int {
        return java.lang.Long.hashCode(tid) * 31 + columnId
    }
}
